at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -46,75 +50,112 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
   }
 
   @override
-  Future<SyncResult> sync({Function? onDone, Function? onError}) async {
-    SyncResult syncResult;
-    // If sync in-progress, return.
-    if (_isSyncInProgress) {
-      _logger.info('Sync already in-progress. Cannot start a new sync');
-      syncResult = SyncResult();
-      syncResult.syncStatus = SyncStatus.failure;
-      syncResult.atClientException = AtClientException(
-          'AT0014', 'Sync-InProgress. Cannot start a new sync process');
-      if (onError != null) {
-        onError(syncResult);
-      }
-      return syncResult;
+  void sync({Function? onDone, Function? onError}) {
+    final syncRequest = SyncRequest();
+    syncRequest.onDone = onDone;
+    syncRequest.onError = onError;
+    syncRequest.requestedOn = DateTime.now().toUtc();
+    syncRequest.result = SyncResult();
+    _syncRequests.add(syncRequest);
+    //#TODO should we call _processQueue here or in cron or in _statsServiceListener ?
+    if (_syncRequests.length > _syncRequestThreshold ||
```

Comment:
Not here. This should happen in a cron even when no one is calling sync.

Reply:
done